source/slang/slang-lower-to-ir.cpp

```diff
@@ -986,6 +998,8 @@ IRStructKey* getInterfaceRequirementKey(
     IRGenContext*   context,
     Decl*           requirementDecl)
 {
+    if (auto genericDecl = as<GenericDecl>(requirementDecl))
+        return getInterfaceRequirementKey(context, genericDecl->inner);
```

Comment:
This doesn't seem right. Why did you need to make this edit?

For a generic interface requirement, the `GenericDecl` itself is the requirement, not whatever its inner declaration is, right?

Reply:
It seems that `getInterfaceRequirementKey` is called on the generic decl when using an interface type (e.g. when generating the `lookup_interface_method` inst, and called on the inner function decl when lowering the interface type, thus two duplicate keys with the same name will be created and the lookup will fail.

Reply:
That's a bit unfortunate, and it would be nice to figure out if we can make the cases more consistent without a workaround like this, but if the workaround prevents a bug for now then I'm fine with it. Maybe it is worth adding a comment to this conditional to be clear about why you had to add it?

Reply:
This stems from that `WitnessTable` are constructed from the inner decls in `slang-check-decl.cpp`. Making it consistent would require changing `doesGenericSignatureMatchRequirement`, currently `doesGenericSignatureMatchRequirement` just calls `doesMemberSignatureMatchRequirement(genDecl->inner)` so the AST witness table will always contain the inner decl as the requirement key. Not sure what would be the cleanest way to fix this.

Reply:
I added a comment for this issue.